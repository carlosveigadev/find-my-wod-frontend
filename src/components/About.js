import {
  Box, Flex, Heading, Text, Center,
} from '@chakra-ui/react';
import { useHistory } from 'react-router-dom';
import { AiOutlineBackward } from 'react-icons/ai';

const About = () => {
  const history = useHistory();
  return (
    <Flex direction="column" m="2em">
      <Center>
        <button
          type="button"
          onClick={() => {
            history.goBack();
          }}
        >
          <AiOutlineBackward fontSize="2em" color="darkgray" />
        </button>
      </Center>

      <Heading as="h1" size="xl" mt="1em">Instructions</Heading>
      <Box mt="2em" mb="1em">
        <Heading as="h2" size="md">Workout of the day (WOD):</Heading>
        <Text>
          Plain and simple, this is the set of modalities that
          your coach uses to put you
          through hell on any given day.
        </Text>
      </Box>
      <Box my="1em">
        <Heading as="h2" size="md">As many rounds as possible (AMRAP):</Heading>
        <Text>
          Complete a circuit as many times as you can
          within a given time frame. For example,
          six-minute AMRAP: 5 deadlifts,
          10 pullups, 20 double unders. When the six minutes is up, record your
          total rounds completed. Note: AMRAP may also mean
          “as many reps as possible.”
        </Text>
      </Box>
      <Box my="1em">
        <Heading as="h2" size="md">MetCon:</Heading>
        <Text>
          Short for metabolic conditioning, this CrossFit devil
          is usually a few exercises repeated AMRAP-style. The “Cindy”
          workout of the day (20 minutes of 5 pullups, 10 pushups, and 15 squats)
          is a good example. Some boxes offer MetCon-only classes for anybody
          who wants to steer clear of the heavy lifting associated with standard CrossFit WODs.
        </Text>
      </Box>
    </Flex>
  );
};

export default About;
