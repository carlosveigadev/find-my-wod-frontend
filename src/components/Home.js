import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Navbar from './Navbar';

const Home = ({ isLoggedIn }) => {
  if (isLoggedIn) {
    return (
      <>
        <Navbar />
        <h1>
          Hi, here you can check all the Wods available.
        </h1>
      </>
    );
  }
  return (
    <>
      <Link to={{ pathname: '/login' }}>Log In</Link>
      <Link to={{ pathname: '/signin' }}>Sign In</Link>
    </>
  );
};

const mapStateToProps = state => ({
  isLoggedIn: state.user.isLoggedIn,
});

Home.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
};

export default connect(mapStateToProps)(Home);
