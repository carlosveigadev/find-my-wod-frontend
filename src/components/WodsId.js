import PropTypes from 'prop-types';

const WodsId = ({ location }) => {
  const { wod } = location.state;
  return (
    <>
      <h1>{wod.title}</h1>
      <h2>{wod.description}</h2>
      <h2>{wod.image}</h2>
    </>
  );
};

WodsId.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default WodsId;
