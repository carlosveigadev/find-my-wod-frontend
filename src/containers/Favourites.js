import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, useHistory } from 'react-router-dom';

const Favourites = ({ favouriteWods }) => {
  const history = useHistory();
  if (favouriteWods.length !== 0) {
    return (
      <>
        {favouriteWods.map(wod => <Link key={wod.id} to={{ pathname: `/wods/${wod.id}`, state: { wod } }}>{wod.title}</Link>)}
        <button
          type="button"
          onClick={() => {
            history.goBack();
          }}
        >
          Go back
        </button>
      </>
    );
  }
  return (
    <>
      <h1>You have no favorite wods...</h1>
      <h1>
        Click
        {' '}
        <Link to="/">Here</Link>
        {' '}
        to go trough the Dashboard.
        {' '}
      </h1>
    </>
  );
};

Favourites.propTypes = {
  favouriteWods: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = state => ({
  favouriteWods: state.favouriteStore,
});

export default connect(mapStateToProps, null)(Favourites);
