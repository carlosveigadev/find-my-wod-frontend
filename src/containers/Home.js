import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { useRef } from 'react';
import { Link } from 'react-router-dom';
import {
  Center,
  Flex,
  Box,
  Text,
  useDisclosure,
  Drawer,
  DrawerOverlay,
  DrawerContent,
  DrawerBody,
  DrawerCloseButton,
  Heading,
} from '@chakra-ui/react';
import { GiHamburgerMenu } from 'react-icons/gi';
import Navbar from '../components/Navbar';
import Wods from './Wods';
import style from '../assets/css/Home.module.css';

const Home = ({ isLoggedIn }) => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  const btnRef = useRef();
  if (isLoggedIn) {
    return (
      <>
        <button className={style.hamburguerIcon} type="button" ref={btnRef} onClick={onOpen}>
          <GiHamburgerMenu />
        </button>
        <Drawer
          isOpen={isOpen}
          placement="left"
          onClose={onClose}
          finalFocusRef={btnRef}
        >
          <DrawerOverlay>
            <DrawerContent>
              <DrawerCloseButton color="darkgray" />
              <DrawerBody>
                <Navbar />
              </DrawerBody>
            </DrawerContent>
          </DrawerOverlay>
        </Drawer>
        <Center py="0.75em">
          <Heading>WODs</Heading>
        </Center>
        <Wods />
      </>
    );
  } return (
    <Box
      bgSize="100% 100%"
      bgPosition="Center"
      bgImage="url(../assets/images/logo.jpg)"
      bgRepeat="no-repeat"
    >
      <Box bgColor="rgb(255, 255, 255, 0.65)">
        <Center color="black">
          <Flex flexDirection="column" h="100vh" pt="35vh">
            <Box py="1em">
              <Text align="center" fontSize="xl">Welcome to Find my Wod!</Text>
              <Text align="center" fontSize="xs">Choose an option to start.</Text>
            </Box>
            <Link className={style.button} to={{ pathname: '/login' }}> Log In </Link>
            <Link className={style.button} to={{ pathname: '/signin' }}>Sign Up</Link>
          </Flex>
        </Center>
      </Box>
    </Box>
  );
};

const mapStateToProps = state => ({
  isLoggedIn: state.userStore.isLoggedIn,
});

Home.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
};

export default connect(mapStateToProps)(Home);
