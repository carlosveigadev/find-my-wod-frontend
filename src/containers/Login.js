import { useState } from 'react';
import { logInUser } from '../api-requests';

const Login = () => {
  const [state, setState] = useState({
    email: '',
    password: '',
  });

  const handleChange = e => {
    const { id, value } = e.target;
    setState(prevState => ({
      ...prevState,
      [id]: value,
    }));
  };

  const handleSubmit = e => {
    e.preventDefault();
    logInUser(state);
  };

  return (
    <form>
      <h1>Log In</h1>
      <h2>Hello there! Log in to start using Find My Wod.</h2>
      <input type="email" id="email" placeholder="email" required onChange={handleChange} />
      <input type="password" id="password" placeholder="password" required onChange={handleChange} />
      <button type="submit" onClick={handleSubmit}>Log In</button>
    </form>
  );
};

export default Login;
