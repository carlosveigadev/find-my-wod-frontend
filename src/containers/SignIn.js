import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import jwt from 'jwt-decode';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Box, Flex, Text, useToast,
} from '@chakra-ui/react';
import { SignInRequest } from '../api-requests';
import { userData } from '../redux/actions';
import style from '../assets/css/Authorizaton.module.css';

const SignIn = ({ userData }) => {
  const [state, setState] = useState({
    emailSignIn: '',
    passwordSignIn: '',
    password_confirmation: '',
  });

  const handleChange = e => {
    const { id, value } = e.target;
    setState(prevState => ({
      ...prevState,
      [id]: value,
    }));
  };

  const history = useHistory();
  const toast = useToast();

  const handleSubmit = async e => {
    e.preventDefault();
    const data = await SignInRequest(state);
    if (data.statusText === 'Created') {
      const populateReduxStore = {
        isLoggedIn: true,
        userToken: data.data.auth_token,
        userInfo: jwt(data.data.auth_token).email,
      };
      toast({
        position: 'bottom-left',
        title: 'Welcome!',
        description: 'Account Created.',
        status: 'info',
        duration: 3000,
        isClosable: true,
      });
      userData(populateReduxStore);
      history.push('/');
    } else {
      toast({
        position: 'bottom-left',
        title: 'Something went wrong!',
        description: 'Sorry, refresh the page and try again.',
        status: 'error',
        duration: 3000,
        isClosable: true,
      });
    }
  };

  return (
    <Box
      bgSize="100% 100%"
      bgPosition="Center"
      bgImage="url(../assets/images/logo.jpg)"
      bgRepeat="no-repeat"
    >
      <Box bgColor="rgb(255, 255, 255, 0.65)">
        <Flex flexDirection="column" h="100vh" pt="35vh">
          <form>
            <Text align="center" fontSize="xl">Sign Up</Text>
            <Text align="center" fontSize="xs">Hello there! Sing in to start using Find My Wod.</Text>
            <input type="email" id="emailSignIn" placeholder="E-mail" required onChange={handleChange} />
            <input type="password" id="passwordSignIn" placeholder="Password" required onChange={handleChange} />
            <input type="password" id="password_confirmation" placeholder="Password Confirmation" required onChange={handleChange} />
            <button className={style.button} type="submit" onClick={handleSubmit}>Sign Up</button>
            <button
              type="button"
              onClick={() => {
                history.push('/login');
              }}
            >
              Log In
            </button>
          </form>
        </Flex>

      </Box>

    </Box>
  );
};

const mapDispatch = {
  userData,
};

SignIn.propTypes = {
  userData: PropTypes.func.isRequired,
};

export default connect(null, mapDispatch)(SignIn);
