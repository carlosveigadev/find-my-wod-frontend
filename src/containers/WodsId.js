import {
  AspectRatio, Center, Flex, Box, Heading, Text,
} from '@chakra-ui/react';
import PropTypes from 'prop-types';
import { AiOutlineBackward } from 'react-icons/ai';
import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { favouriteRequest, fetchFavourites } from '../api-requests';
import { favouriteData } from '../redux/actions';
import style from '../assets/css/WodsId.module.css';

const WodsId = ({
  location, favouriteWods, userToken, favouriteData,
}) => {
  const { wod } = location.state;
  const history = useHistory();

  const ids = [];
  favouriteWods.map(item => ids.push(item.id));

  const handleSubmit = async e => {
    const status = e.target.id;
    await favouriteRequest(wod.id, status, userToken);
    const newData = await fetchFavourites(userToken);
    favouriteData(newData);
  };

  const video = `https://www.youtube.com/embed/${wod.image}`;

  return (
    <Flex direction="column" mt="2em" alignItems="center">
      <Center>
        <button
          type="button"
          onClick={() => {
            history.goBack();
          }}
        >
          <AiOutlineBackward fontSize="2em" color="darkgray" />
        </button>
      </Center>
      <Box mt="2em">
        <AspectRatio maxW="440px" ratio={4 / 3}>
          <iframe
            title={wod.title}
            src={video}
            allowFullScreen
          />
        </AspectRatio>
        <Box mx="2em" mt="1em">
          <Heading s="h1" size="lg">{wod.title}</Heading>
          <Text my="1em">{wod.description}</Text>
          {ids.includes(wod.id)
            ? (
              <button className={style.button} type="button" id="unfavourite" onClick={handleSubmit}>
                Remove from Favourites
              </button>
            )
            : <button className={style.button} type="button" id="favourite" onClick={handleSubmit}>Add to Favourites</button>}
        </Box>
      </Box>
    </Flex>
  );
};

WodsId.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
  favouriteWods: PropTypes.arrayOf(PropTypes.object).isRequired,
  userToken: PropTypes.string.isRequired,
  favouriteData: PropTypes.func.isRequired,
};

const mapDispatch = {
  favouriteData,
};

const mapStateToProps = state => ({
  favouriteWods: state.favouriteStore,
  userToken: state.userStore.userToken,
});

export default connect(mapStateToProps, mapDispatch)(WodsId);
