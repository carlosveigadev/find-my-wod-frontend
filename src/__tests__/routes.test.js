import React from 'react';
import renderer from 'react-test-renderer';
import { createStore, combineReducers } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import userReducer from '../redux/reducers/user';
import favouriteReducer from '../redux/reducers/favourites';
import Home from '../components/Home';
import Login from '../containers/Login';
import WodsId from '../containers/WodsId';
import Favourites from '../containers/Favourites';
import About from '../components/About';
import SignIn from '../containers/SignIn';

describe('Check if routes work ', () => {
  it('renders Home page correctly in the proper route.', () => {
    const rootReducer = combineReducers(
      { userStore: userReducer, favouriteStore: favouriteReducer },
    );
    const store = createStore(rootReducer);

    const home = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });

  it('renders About page correctly in the proper route.', () => {
    const about = renderer
      .create(
        <BrowserRouter>
          <About />
        </BrowserRouter>,
      )
      .toJSON();
    expect(about).toMatchSnapshot();
  });

  it('renders Login page correctly in the proper route.', () => {
    const rootReducer = combineReducers(
      { userStore: userReducer, favouriteStore: favouriteReducer },
    );
    const store = createStore(rootReducer);
    const login = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(login).toMatchSnapshot();
  });

  it('renders SingUp page correctly in the proper route.', () => {
    const rootReducer = combineReducers(
      { userStore: userReducer, favouriteStore: favouriteReducer },
    );
    const store = createStore(rootReducer);
    const signup = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <SignIn />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(signup).toMatchSnapshot();
  });

  it('renders WodId page correctly in the proper route.', () => {
    const rootReducer = combineReducers(
      { userStore: userReducer, favouriteStore: favouriteReducer },
    );
    const store = createStore(rootReducer);
    const props = { state: { wod: { title: 'test', description: 'test', image: 'test' } } };
    const wodsid = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <WodsId location={props} />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(wodsid).toMatchSnapshot();
  });

  it('renders Favourites page correctly in the proper route.', () => {
    const rootReducer = combineReducers(
      { userStore: userReducer, favouriteStore: favouriteReducer },
    );
    const store = createStore(rootReducer);
    const favourites = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Favourites />
          </BrowserRouter>
        </Provider>,

      )
      .toJSON();
    expect(favourites).toMatchSnapshot();
  });
});
