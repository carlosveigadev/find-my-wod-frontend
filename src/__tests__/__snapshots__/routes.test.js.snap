// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if routes work  renders About page correctly in the proper route. 1`] = `
<div
  className="css-1166ll3"
>
  <div
    className="css-gmuwbf"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        color="darkgray"
        fill="currentColor"
        fontSize="2em"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "darkgray",
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M485.6 249.9L198.2 498c-8.3 7.1-8.3 20.8 0 27.9l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9zm320 0L518.2 498a18.6 18.6 0 0 0-6.2 14c0 5.2 2.1 10.4 6.2 14l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9z"
        />
      </svg>
    </button>
  </div>
  <h1
    className="chakra-heading css-16f1609"
  >
    Instructions
  </h1>
  <div
    className="css-174o6v9"
  >
    <h2
      className="chakra-heading css-0"
    >
      Workout of the day (WOD):
    </h2>
    <p
      className="chakra-text css-0"
    >
      Plain and simple, this is the set of modalities that your coach uses to put you through hell on any given day.
    </p>
  </div>
  <div
    className="css-56tzch"
  >
    <h2
      className="chakra-heading css-0"
    >
      As many rounds as possible (AMRAP):
    </h2>
    <p
      className="chakra-text css-0"
    >
      Complete a circuit as many times as you can within a given time frame. For example, six-minute AMRAP: 5 deadlifts, 10 pullups, 20 double unders. When the six minutes is up, record your total rounds completed. Note: AMRAP may also mean “as many reps as possible.”
    </p>
  </div>
  <div
    className="css-56tzch"
  >
    <h2
      className="chakra-heading css-0"
    >
      MetCon:
    </h2>
    <p
      className="chakra-text css-0"
    >
      Short for metabolic conditioning, this CrossFit devil is usually a few exercises repeated AMRAP-style. The “Cindy” workout of the day (20 minutes of 5 pullups, 10 pushups, and 15 squats) is a good example. Some boxes offer MetCon-only classes for anybody who wants to steer clear of the heavy lifting associated with standard CrossFit WODs.
    </p>
  </div>
</div>
`;

exports[`Check if routes work  renders Favourites page correctly in the proper route. 1`] = `
<div
  className="css-1166ll3"
>
  <div
    className="css-gmuwbf"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        color="darkgray"
        fill="currentColor"
        fontSize="2em"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "darkgray",
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M485.6 249.9L198.2 498c-8.3 7.1-8.3 20.8 0 27.9l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9zm320 0L518.2 498a18.6 18.6 0 0 0-6.2 14c0 5.2 2.1 10.4 6.2 14l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9z"
        />
      </svg>
    </button>
  </div>
  <h1
    className="chakra-heading css-16f1609"
  >
    Favourite WODs
  </h1>
  <div
    className="css-1v8nkjm"
  >
    <span
      className="css-z730za"
    >
      <a
        href="/wods/undefined"
        onClick={[Function]}
      />
    </span>
  </div>
</div>
`;

exports[`Check if routes work  renders Home page correctly in the proper route. 1`] = `
<div
  className="css-128chhw"
>
  <div
    className="css-rn9ki1"
  >
    <div
      className="css-1gy61cu"
    >
      <div
        className="css-26xpnt"
      >
        <div
          className="css-kae6bt"
        >
          <p
            className="chakra-text css-1vhs2ha"
          >
            Welcome to Find my Wod!
          </p>
          <p
            className="chakra-text css-uq0a7f"
          >
            Choose an option to start.
          </p>
        </div>
        <a
          className="button"
          href="/login"
          onClick={[Function]}
        >
           Log In 
        </a>
        <a
          className="button"
          href="/signin"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Check if routes work  renders Login page correctly in the proper route. 1`] = `
<div
  className="css-128chhw"
>
  <div
    className="css-rn9ki1"
  >
    <div
      className="css-26xpnt"
    >
      <form>
        <p
          className="chakra-text css-1vhs2ha"
        >
          Log In
        </p>
        <p
          className="chakra-text css-uq0a7f"
        >
          Hello there! Log in to start using Find My Wod.
        </p>
        <input
          id="email"
          onChange={[Function]}
          placeholder="E-mail"
          required={true}
          type="email"
        />
        <input
          id="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
        <button
          className="button"
          onClick={[Function]}
          type="submit"
        >
          Log In
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Sign Up
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Check if routes work  renders SingUp page correctly in the proper route. 1`] = `
<div
  className="css-128chhw"
>
  <div
    className="css-rn9ki1"
  >
    <div
      className="css-26xpnt"
    >
      <form>
        <p
          className="chakra-text css-1vhs2ha"
        >
          Sign Up
        </p>
        <p
          className="chakra-text css-uq0a7f"
        >
          Hello there! Sing in to start using Find My Wod.
        </p>
        <input
          id="emailSignIn"
          onChange={[Function]}
          placeholder="E-mail"
          required={true}
          type="email"
        />
        <input
          id="passwordSignIn"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
        <input
          id="password_confirmation"
          onChange={[Function]}
          placeholder="Password Confirmation"
          required={true}
          type="password"
        />
        <button
          className="button"
          onClick={[Function]}
          type="submit"
        >
          Sign In
        </button>
      </form>
    </div>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Log In
  </button>
</div>
`;

exports[`Check if routes work  renders WodId page correctly in the proper route. 1`] = `
<div
  className="css-1eb1udp"
>
  <div
    className="css-gmuwbf"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        color="darkgray"
        fill="currentColor"
        fontSize="2em"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "darkgray",
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M485.6 249.9L198.2 498c-8.3 7.1-8.3 20.8 0 27.9l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9zm320 0L518.2 498a18.6 18.6 0 0 0-6.2 14c0 5.2 2.1 10.4 6.2 14l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9z"
        />
      </svg>
    </button>
  </div>
  <div
    className="css-1v8nkjm"
  >
    <div
      className="chakra-aspect-ratio css-103ady7"
    >
      <iframe
        allowFullScreen={true}
        src="https://www.youtube.com/embed/test"
        title="test"
      />
    </div>
    <div
      className="css-759ns7"
    >
      <h2
        className="chakra-heading css-0"
        s="h1"
      >
        test
      </h2>
      <p
        className="chakra-text css-56tzch"
      >
        test
      </p>
      <button
        className="button"
        id="unfavourite"
        onClick={[Function]}
        type="button"
      >
        Remove from Favourites
      </button>
    </div>
  </div>
</div>
`;
